src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -254,11 +330,259 @@ void CableModel<DataTypes>::checkIndicesRegardingState()
+        {
+            DistanceToPoint top = *queue.begin();
+            queue.erase(queue.begin());
+            BaseMeshTopology::PointID v = top.second; 
+            const vector<BaseMeshTopology::PointID> neighboors = l_surfaceTopology.get()->getVerticesAroundVertex(v);
+            for (unsigned int j=0 ; j<neighboors.size(); ++j)
+            {
+                sofa::core::topology::BaseMeshTopology::PointID vn = neighboors[j];
+
+                Coord pv; 
+                getPositionFromTopology(pv, v);
+                Coord pvn;
+                getPositionFromTopology(pvn, vn);
+                Real d = 0.0;
+                if (id_method == 2)
+                {
+                    d = distances[v] + (pv - pvn).norm();
+                }
+                else if (id_method == 1)
+                {
+                    d = (pvn - cablePositions[indices[i]]).norm();
+                }
+                if((distances[vn]) > d)
+                {
+                    qit=queue.find(DistanceToPoint(distances[vn],vn));
+                    if(qit != queue.end()) {queue.erase(qit);}
+                    distances[vn] = d;
+                    queue.insert(DistanceToPoint(d,vn) );
+                }
+            }
+        }
+
+        // Distribute forces
+        m_totalRatio[i] = 0.0;
+        for(unsigned int p=0; p<distances.size(); p++)
+        {
+            if(distances[p] < maxDistance[i])
+            {
+                const auto ratio = rabs(distances[p] - maxDistance[i]);
+
+                m_totalRatio[i] += ratio;
+                m_ratios[i].push_back(ratio);
+                m_areaIndices[i].push_back(p);
+            } 
+        }
+
+        // Normalize force ratios
+        for(unsigned int j=0; j<m_ratios[i].size(); j++)
+        {
+            if (m_totalRatio[i] == 0.0)
+            {
+                m_ratios[i][j] = 0.0;
+            }
+            else 
+            {
+                m_ratios[i][j] = m_ratios[i][j] / m_totalRatio[i];
+            }
+        }
+            
+    }
+}
+
+
+template<class DataTypes>
+unsigned int CableModel<DataTypes>::computeClosestIndice(const Coord& position)
+{
+    Coord p0;
+    getPositionFromTopology(p0, 0);
+    auto closest_distance = (position - p0).norm();
+    unsigned int closest_vertice = 0;
+    for(unsigned int j=1; j<l_surfaceTopology.get()->getNbPoints(); j++)
+    {
+        Coord pj;
+        getPositionFromTopology(pj, j);
+        const auto distance = (position - pj).norm();
+        if(distance < closest_distance) 
+        {
+            closest_distance = distance;
+            closest_vertice = j;
+        }
+    } 
+   return closest_vertice;
+}
+
+template<class DataTypes>
+void CableModel<DataTypes>::getPositionFromTopology(Coord& position, sofa::Index index)
+{
+}
+
+template<class DataTypes>
+SReal CableModel<DataTypes>::getDistanceToTriangle(const Coord& position, const Triangle& triangle, Coord& projectionOnTriangle)
+{
+}
```

Comment:
Does it compile if you remove those lines?
```suggestion
```

Reply:
Yes, I removed them.